pop-api/examples/fungibles/lib.rs

```diff
@@ -1,108 +1,281 @@
 #![cfg_attr(not(feature = "std"), no_std, no_main)]
 
-use ink::prelude::vec::Vec;
+use ink::prelude::{string::String, vec::Vec};
 use pop_api::{
-	fungibles::{self as api},
 	primitives::TokenId,
-	StatusCode,
+	v0::fungibles::{
+		self as api,
+		events::{Approval, Created, Transfer},
+		traits::{Psp22, Psp22Burnable, Psp22Metadata, Psp22Mintable},
+		PSP22Error,
+	},
 };
 
-pub type Result<T> = core::result::Result<T, StatusCode>;
+#[cfg(test)]
+mod tests;
+#[cfg(test)]
+mod utils;
 
 #[ink::contract]
 mod fungibles {
 	use super::*;
 
 	#[ink(storage)]
-	#[derive(Default)]
-	pub struct Fungibles;
+	pub struct Fungibles {
+		id: TokenId,
+	}
 
 	impl Fungibles {
+		fn emit_created_event(&mut self, id: u32, creator: AccountId, admin: AccountId) {
+			self.env().emit_event(Created { id, creator, admin });
+		}
+
+		fn emit_transfer_event(
+			&mut self,
+			from: Option<AccountId>,
+			to: Option<AccountId>,
+			value: Balance,
+		) {
+			self.env().emit_event(Transfer { from, to, value });
+		}
+
+		fn emit_approval_event(&mut self, owner: AccountId, spender: AccountId, value: Balance) {
+			self.env().emit_event(Approval { owner, spender, value });
+		}
+
+		/// Instantiate the contract and wrap around an existing token.
+		///
+		/// # Parameters
+		/// * - `token` - The token.
 		#[ink(constructor, payable)]
-		pub fn new() -> Self {
-			Default::default()
+		pub fn new_existing(id: TokenId) -> Result<Self, PSP22Error> {
+			// Make sure token exists.
+			if !api::token_exists(id).unwrap_or_default() {
+				return Err(PSP22Error::Custom(String::from("Unknown")));
+			}
+			let contract = Self { id };
+			Ok(contract)
 		}
 
-		#[ink(message)]
-		pub fn total_supply(&self, token: TokenId) -> Result<Balance> {
-			api::total_supply(token)
+		/// Instantiate the contract and create a new token. The token identifier will be stored
+		/// in contract's storage.
+		///
+		/// # Parameters
+		/// * - `id` - The identifier of the token.
+		/// * - `admin` - The account that will administer the token.
+		/// * - `min_balance` - The minimum balance required for accounts holding this token.
+		#[ink(constructor, payable)]
+		pub fn new(
+			id: TokenId,
+			// TODO: If admin is different than the contract address, `NoPermission` thrown for mint, burn
```

Comment:
I prefer removing the parameter for now. Because even we support developers to update the admin with `set_team` api, the `admin` can't use method like `mint` and `burn` as mentioned in https://github.com/r0gue-io/pop-node/pull/301

Reply:
Not if the contract is not set as `admin`*